### ISSUE TITLE
bazel build error : in fail_rule rule @local_jdk//:jdk:

### ISSUE THREAD

**Issue by pekochun** [google/mediapipe#2772]
<em>Please make sure that this is a build/installation issue and also refer to the [troubleshooting](https://google.github.io/mediapipe/getting_started/troubleshooting.html) documentation before raising any issues.</em>

**System information** (Please provide as much relevant information as possible)
- OS Platform and Distribution 
I use macOS Big Sur version 11.5.2

- Compiler version (e.g. gcc/g++ 8 /Apple clang version 12.0.0):
Mac terminal

- Programming Language and version ( e.g. C++ 14, Python 3.6, Java ):
bazel

- [MediaPipe version](https://github.com/google/mediapipe/releases):
I didn't know how to look it up.

- Bazel version:
Bazelisk version: development
Build label: 3.7.2
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Thu Dec 17 17:02:20 2020 (1608224540)
Build timestamp: 1608224540
Build timestamp as int: 1608224540

- XCode and Tulsi versions (if iOS):
Xcode13.0

- Android SDK and NDK versions (if android):
Android SDK 31.00
Android NDK 21.47075529

- Android [AAR](https://google.github.io/mediapipe/getting_started/android_archive_library.html) ( if android):
I don't know what this means.

- OpenCV version (if running on desktop):
I didn't know how to look it up.

**Describe the problem**:
I followed the tutorial below to configure bazel.
However, I got an error.
I will show the error message later.
https://google.github.io/mediapipe/getting_started/android.html

I tried the following command.
$ bazel build -c opt --config=android_arm64 

**Complete Logs:**
The error log is as follows
There are many errors, but I cannot find the cause.

Mac-mini:mediapipe mac$ bazel build -c opt --config=android_arm64 mediapipe/examples/android/src/java/com/google/mediapipe/apps/handtrackinggpu:handtrackinggpu
DEBUG: Rule 'rules_foreign_cc' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "c2cdcf55ffaf49366725639e45dedd449b8c3fe22b54e31625eb80ce3a240f1e"
DEBUG: Repository rules_foreign_cc instantiated at:
  /Users/mac/Documents/mediapipe-master/mediapipe/mediapipe/WORKSPACE:42:13: in <toplevel>
Repository rule http_archive defined at:
  /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'com_google_absl' because it already exists.
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'com_google_protobuf' because it already exists.
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'com_google_googletest' because it already exists.
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'com_github_gflags_gflags' because it already exists.
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'build_bazel_rules_apple' because it already exists.
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'build_bazel_rules_swift' because it already exists.
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'build_bazel_apple_support' because it already exists.
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'bazel_skylib' because it already exists.
DEBUG: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/org_tensorflow/third_party/repo.bzl:109:14: 
Warning: skipping import of repository 'pybind11' because it already exists.
INFO: Build options --compilation_mode, --cpu, --crosstool_top, and 3 more have changed, discarding analysis cache.
DEBUG: Rule 'rules_cc' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "3839996049629e6377abdfd04681ddeeb0cc3db13b9d2ff81bf46700cb4529f7"
DEBUG: Repository rules_cc instantiated at:
  /Users/mac/Documents/mediapipe-master/mediapipe/mediapipe/WORKSPACE:36:13: in <toplevel>
Repository rule http_archive defined at:
  /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>
WARNING: API level 30 specified by android_ndk_repository 'androidndk' is not available. Using latest known API level 29
ERROR: /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/local_jdk/BUILD.bazel:3:10: in fail_rule rule @local_jdk//:jdk: 
Traceback (most recent call last):
	File "/private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/bazel_tools/tools/jdk/fail_rule.bzl", line 19, column 13, in _fail_rule_impl
		fail("%s %s" % (ctx.attr.header, ctx.attr.message))
Error in fail: Auto-Configuration Error: Cannot find Java binary bin/java in /var/tmp/_bazel_mac/install/e5700781a43090e74aa4012c7e525e2a/embedded_tools/tools/jdk/nosystemjdk; either correct your JAVA_HOME, PATH or specify embedded Java (e.g. --javabase=@bazel_tools//tools/jdk:remote_jdk11)
INFO: Repository remote_java_tools_darwin instantiated at:
  /DEFAULT.WORKSPACE.SUFFIX:259:6: in <toplevel>
  /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/bazel_tools/tools/build_defs/repo/utils.bzl:201:18: in maybe
Repository rule http_archive defined at:
  /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>
INFO: Repository remotejdk11_macos instantiated at:
  /DEFAULT.WORKSPACE.SUFFIX:181:6: in <toplevel>
  /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/bazel_tools/tools/build_defs/repo/utils.bzl:201:18: in maybe
Repository rule http_archive defined at:
  /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>
INFO: Repository maven instantiated at:
  /Users/mac/Documents/mediapipe-master/mediapipe/mediapipe/WORKSPACE:331:14: in <toplevel>
  /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/rules_jvm_external/defs.bzl:105:19: in maven_install
Repository rule coursier_fetch defined at:
  /private/var/tmp/_bazel_mac/ffbac629f6d5b918d7747960b47b6ab6/external/rules_jvm_external/coursier.bzl:1009:33: in <toplevel>
ERROR: Analysis of target '//mediapipe/examples/android/src/java/com/google/mediapipe/apps/handtrackinggpu:handtrackinggpu' failed; build aborted: Analysis of target '@local_jdk//:jdk' failed
INFO: Elapsed time: 2.028s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (15 packages loaded, 7328 targets configured)

**Comment by sgowroji** [comment]
Hi @pekochun, Did you set the Java SDK, NDK environment variable and verify the above .  Error from the log 
`Error in fail: Auto-Configuration Error: Cannot find Java binary bin/java in /var/tmp/_bazel_mac/install/e5700781a43090e74aa4012c7e525e2a/embedded_tools/tools/jdk/nosystemjdk; either correct your JAVA_HOME, PATH or specify embedded Java (e.g. --javabase=@bazel_tools//tools/jdk:remote_jdk11)`

**Comment by pekochun** [comment]
Thanks for the reply.
What is the path of Java SDK,NDK, is it the same as the path of Android Studio SDK,NDK?

I already have the following set up.
export ANDROID_HOME=/Users/mac/Library/Android/sdk
export ANDROID_NDK_HOME=/Users/mac/Library/Android/sdk/ndk/21.4.7075529

I thought it might be possible that I had not installed Java, so I installed Java.
Then I ran the command.
Then I got a slightly different error than the last time.
